src/probability/stopping.lean

```diff
@@ -211,26 +227,155 @@ def adapted (f : filtration ι m) (u : ι → α → β) : Prop :=
 
 namespace adapted
 
-lemma add [has_add β] [has_measurable_add₂ β] {u v : ι → α → β} {f : filtration ι m}
-  (hu : adapted f u) (hv : adapted f v) : adapted f (u + v) :=
-λ i, @measurable.add _ _ _ _ (f i) _ _ _ (hu i) (hv i)
+lemma add [has_add β] [has_measurable_add₂ β] (hu : adapted f u) (hv : adapted f v) :
+  adapted f (u + v) :=
+λ i, (hu i).add (hv i)
 
-lemma neg [has_neg β] [has_measurable_neg β] {u : ι → α → β} {f : filtration ι m}
-  (hu : adapted f u) : adapted f (-u) :=
-λ i, @measurable.neg _ α _ _ _ (f i) _ (hu i)
+lemma neg [has_neg β] [has_measurable_neg β] (hu : adapted f u) : adapted f (-u) :=
+λ i, (hu i).neg
 
-lemma smul [has_scalar ℝ β] [has_measurable_smul ℝ β] {u : ι → α → β} {f : filtration ι m}
-  (c : ℝ) (hu : adapted f u) : adapted f (c • u) :=
-λ i, @measurable.const_smul ℝ β α _ _ _ (f i) _ _ (hu i) c
+lemma smul [has_scalar ℝ β] [has_measurable_smul ℝ β] (c : ℝ) (hu : adapted f u) :
+  adapted f (c • u) :=
+λ i, (hu i).const_smul c
 
 end adapted
 
 variable (β)
-
 lemma adapted_zero [has_zero β] (f : filtration ι m) : adapted f (0 : ι → α → β) :=
-λ i, @measurable_zero β α (f i) _ _
+λ i, @measurable_zero β α _ (f i) _
+variable {β}
 
-end adapted
+/-- Progressively measurable process. A sequence of functions `u` is said to be progressively
+measurable with respect to a filtration `f` if at each point in time `i`, `u` restricted to
+`set.Iic i × α` is measurable with respect to the product `measurable_space` structure where the
+σ-algebra used for `α` is `f i`.
+The usual definition uses the interval `[0,i]`, which we replace by `set.Iic i`. We recover the
+usual definition for index types `ℝ≥0` or `ℕ`. -/
+def prog_measurable [measurable_space ι] (f : filtration ι m) (u : ι → α → β) : Prop :=
+∀ i, measurable[prod.measurable_space' subtype.measurable_space (f i)]
```

Comment:
Would it be a good idea to rename `prod.measurable_space'` to `measurable_space.prod` to take advantage of dot notation

Reply:
Good idea! Thanks.